client/src/lib/components/zzz/ActionButton.svelte

```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+    import { cn } from '$lib/utils/cn';
+
+    const { class: externalClass = '', action, disabled } = $props();
+</script>
+
+<button
+    {disabled}
+    class={cn(externalClass, 'btn btn-circle btn-lg z-10 ring-offset-neutral', {
+        'btn-info': action === 'Zip',
+        'btn-success': action === 'Zap',
+        'btn-warning': action === 'Zop',
+    })}
```

Comment:
I would typically prefer a `function` with a `switch`-`return` here instead of redundantly checking three mutually exclusive conditions.